logstash-core/lib/logstash/util.rb

```diff
@@ -183,4 +183,21 @@ def self.stringify_symbols(o)
       o
     end
   end
+
+  def self.deep_clone(o)
+    case o
+    when Hash
+      o.inject({}) {|h, (k,v)| h[k] = deep_clone(v); h }
+    when Array
+      o.map {|v| deep_clone(v) }
+    when Fixnum, Symbol, IO, TrueClass, FalseClass, NilClass
+      o
+    when LogStash::Codecs::Base
+      o.clone
+    when String
+      o.dup
```

Comment:
i wonder why dup and not clone? I understand for String would have the same effect, right?